server/routerlicious/packages/lambdas/src/utils/checkpointHelper.ts

```diff
@@ -31,3 +32,52 @@ export interface ICheckpoint {
 	// time in milliseconds since the last checkpoint
 	lastCheckpointTime: number;
 }
+
+export async function restoreFromCheckpoint(documentId: string, tenantId:string, service: string, localCheckpointEnabled: boolean, checkpointRepository: ICheckpointRepository, document: IDocument): Promise<any> {
+    let checkpoint;
+    let lastCheckpoint: IDeliState;
+    let isLocalCheckpoint = false;
+    const restoreFromCheckpointMetric = Lumberjack.newLumberMetric(LumberEventName.RestoreFromCheckpoint);
+    let checkpointSource = "defaultGlobalCollection";
+
+    if (!localCheckpointEnabled || !checkpointRepository){
+        // If we cannot checkpoint locally, use document
+        lastCheckpoint = JSON.parse(document[service]);
+    } else {
+        // Search checkpoints collection for checkpoint
+        checkpoint = await checkpointRepository.getCheckpoint(documentId, tenantId)
+        .catch((error) => {
+            Lumberjack.error(`Error retrieving local checkpoint`, getLumberBaseProperties(documentId, tenantId));
+            checkpointSource = "notFoundInLocalCollection";
+        });
+
+        if(checkpoint?.deli) {
+            lastCheckpoint = JSON.parse(checkpoint[service]);
+            checkpointSource = "foundInLocalCollection";
+            isLocalCheckpoint = true;
+        } else {
+            // If checkpoint does not exist, use document
+            checkpointSource = "notFoundInLocalCollection";
+            lastCheckpoint = JSON.parse(document[service]);
+        }
+
+    }
+
+    restoreFromCheckpointMetric.setProperties({
+        [BaseTelemetryProperties.tenantId]: tenantId,
+        [BaseTelemetryProperties.documentId]: documentId,
+        service,
+        checkpointSource,
+        retrievedFromLocalDatabase: isLocalCheckpoint,
+    });
+
+    if (lastCheckpoint) {
```

Comment:
I think we should do `try catch` for above code to ensure your code flows in here? I know you alrealdy catched the checkpointRepository read part, just in case the JSON.parse or any other throw exceptions?